framework/src/main/java/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -76,6 +79,26 @@ protected void commonAssignmentCheck(
                     getTypeFactory().createIntRangeAnnotation(Range.CHAR_EVERYTHING));
         }
 
+        if (valueTree.getKind() == Kind.METHOD_INVOCATION
+                && TreeUtils.isArrayscopyOfMethodInvocation((MethodInvocationTree) valueTree)
+                && valueType.getKind() == TypeKind.ARRAY) {
+            List<? extends ExpressionTree> args = ((MethodInvocationTree) valueTree).getArguments();
+            AnnotatedTypeMirror arrType = atypeFactory.getAnnotatedType(args.get(0));
+            if (TreeUtils.isArrayLengthAccess(args.get(1))) {
```

Comment:
This only tests that the second argument is any array length access expression. The correct test needs to check that it is an array length access for the array that is the first argument. You can add a test case for this:

```
    public static void m4(String @MinLen(1) [] args, String[] otherArray) {
        // :: error: assignment.type.incompatible
        String @MinLen(1) [] args2 = Arrays.copyOf(args, otherArray.length);
    }
```

Reply:
Do we want to check this now? As suggested above I have added to `valuetype` any `@IntVal` or `@IntRange` annotation on the second argument. Suppose the user intends to have `args2` of the same length as `otherArray`, then this check might lead to a false positive.

Reply:
In the test case I suggested, that is a true positive: `otherArray` might not be `@MinLen(1)`, even if the user hopes that it is - its length could be zero. I think you can add that test method verbatim into the existing test (except for changing the name).